### DISCUSSION TITLE
Consistent star-rating (quality score) or better explanation of its meaning

### DISCUSSION THREAD

**Discussion by alexbarcelo** [photoprism/photoprism#2569] [General]
After looking around, I still am a bit puzzled on the meaning of the "quality score" going from 1 to 7 stars. (The quality score can be seen in a 7-star grading on the "gear tab", while in "edit mode" of a single photo).

I can only assume that this is related to the 7 levels of rating according to the [IPTC image rating field](https://iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#image-rating) which coincidentally (not really) matches what Darktable uses and I can only assume that Lightroom and/or others use it too. And that is really a 5-star rating system plus two additional special cases for "rejected" and "unrated" photos.

If that's the case, then the Photoprism way of showing it is a bit strange. I like the minimalism of darktable that uses two distinct icons for the first two levels:
![Darktable icons for rating](https://user-images.githubusercontent.com/665750/182022252-14541c4d-4cd3-4a3b-9693-e764206b67c8.png)

In case I am completely mistaken, then I apologize and either I am blind and I have skipped some place on the documentation where this is explained or the documentation misses that (why 7 levels?). I am also troubled on the meaning of quality score explained in the [review part of the documentation](https://docs.photoprism.app/user-guide/organize/review/). Does the quality score change when metadata is added? Does the quality score change when you "heart" it? Are those the same? Which takes precedence?

I see that there is an issue discussing the implementation of 5 star rating (#713). It is not the same as quality score? I plan to import all my darktable-tagged library onto Photoprism, and I assumed that the quality score is the same as rating and it may be taken from the EXIF / XMP / IPTC / whatever metadata the photo has. But after reading a little bit more on the documentation and GitHub issues I am now more confused than before and maybe it was all a big misunderstanding on my part.

**Comment by alexbarcelo** [comment]
I apologize to @lastzero for interrupting vacations (I didn't want to label it as a bug when I posted it as #2568, but there are not really any other options, maybe I trespassed a little bit, I felt that it was not a discussion at the time, sorry).

I still fail to understand the internal point system. I hope somebody in the community can point me to the source of this 7 levels. The documentation is somewhat fuzzy and I don't understand the quality score and the relation with the heart / favorite mechanism.

**Reply by lastzero** [reply]
https://github.com/photoprism/photoprism/blob/0e1222c8309c14c34443631954f29be4ef1f7f55/internal/entity/photo_quality.go#L22

**Reply by lastzero** [reply]
It has no relation to IPTC, Exif or the hearts. It's a quality rating based on photo characteristics, not whether you like it or not.
